xonsh/environ.py

```diff
@@ -191,6 +194,7 @@ def xonshconfig(env):
     'PUSHD_SILENT': False,
     'RAISE_SUBPROC_ERROR': False,
     'RIGHT_PROMPT': '',
+    'DYNAMIC_CWD_WIDTH': '20',
```

Comment:
Ok I will use `inf` then. But having float means I won't be able to use integer division as @melund proposed. (As far as I know python does do allow representation of infinity with the integer type)

On the other side I do like the "short-circuit" thing

I don't think it long to compute it's just an overhead of `len(pwd)` integer division + substring exraction.